### PR TITLE
DVC-6132 [fix]: downgrade some JS SDK error logs to warning logs

### PR DESCRIPTION
**Pull request by kaushalkapasi**

### CODE REVIEW

File: sdk/js/src/Client.ts [hunk 1]
```diff
@@ -267,7 +267,7 @@ export class DVCClient implements Client {
 
     track(event: ClientEvent): void {
         if (this._closing) {
-            this.logger.error('Client is closing, cannot track new events.')
+            this.logger.warn('Client is closing, cannot track new events.')
```
Comment (ajwootto) [inline sdk/js/src/Client.ts:270]: this one actually is an error, it's a user problem

File: sdk/js/src/StreamingConnection.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export class StreamingConnection {
             this.onMessage(event.data)
         }
         this.connection.onerror = (err) => {
-            this.logger.error('StreamingConnection error', err)
+            this.logger.warn(`StreamingConnection error ${err}`)
```
Comment (ajwootto) [inline sdk/js/src/StreamingConnection.ts:20]: maybe instead of saying error, say something like "StreamingConnection interrupted, re-establishing.."

Comment (ajwootto) [reply sdk/js/src/StreamingConnection.ts:20]: also according to MDN this situation actually means the connection failed to be established in the first place, not that it was closed later. Are we sure this isn't a problem?
https://developer.mozilla.org/en-US/docs/Web/API/EventSource/error_event

File: sdk/js/src/Client.ts (hunk 1 above)
Comment (ajwootto) [reply sdk/js/src/Client.ts:270]: bump

Comment (kaushalkapasi) [reply sdk/js/src/Client.ts]: just fixed! 😂

**Review by ajwootto — APPROVED** [review]

**Pull request merged**